tests/moduletool/test_add.py

```diff
@@ -0,0 +1,209 @@
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+import os
+import shutil
+from typing import List
+
+import py
+import pytest
+
+from inmanta.command import CLIException
+from inmanta.env import process_env
+from inmanta.module import ModuleV1, ModuleV1Metadata, ModuleV2, Project, ProjectMetadata
+from inmanta.moduletool import ModuleTool
+from packaging.version import Version
+from utils import PipIndex, module_from_template
+
+
+def test_module_add_v1_module_to_project(snippetcompiler_clean) -> None:
+    """
+    Add a V1 module to an inmanta project using the `inmanta module add` command.
+    """
+    project: Project = snippetcompiler_clean.setup_for_snippet(snippet="", autostd=False)
+    requirements_txt_file = os.path.join(project.path, "requirements.txt")
+
+    def _get_content_requirements_txt_file() -> str:
+        with open(requirements_txt_file, "r", encoding="utf-8") as fd:
+            return fd.read()
+
+    def _assert_project_state(constraint: str) -> None:
+        with open(project.get_metadata_file_path(), "r", encoding="utf-8") as fd:
+            project_metadata = ProjectMetadata.parse(fd)
+            assert constraint in project_metadata.requires
+        with open(os.path.join(snippetcompiler_clean.libs, "std", ModuleV1.MODULE_FILE), "r", encoding="utf-8") as fd:
+            module_metadata = ModuleV1Metadata.parse(fd)
+            assert module_metadata.version == constraint.split("==")[1]
+        assert os.listdir(snippetcompiler_clean.libs) == ["std"]
+        assert not _get_content_requirements_txt_file()
+
+    assert not os.listdir(snippetcompiler_clean.libs)
+    assert not _get_content_requirements_txt_file()
+
+    version_constraint = "std==3.0.2"
+    ModuleTool().add(module_req=version_constraint, v1=True, override=False)
+    _assert_project_state(version_constraint)
+
+    new_version_constraint = "std==3.0.1"
+    with pytest.raises(CLIException, match="A dependency on the given module was already defined"):
+        ModuleTool().add(module_req=new_version_constraint, v1=True, override=False)
+    _assert_project_state(version_constraint)
+
+    ModuleTool().add(module_req=new_version_constraint, v1=True, override=True)
+    _assert_project_state(new_version_constraint)
+
+
+@pytest.mark.slowtest
+def test_module_add_v2_module_to_project(
+    tmpdir: py.path.local, snippetcompiler_clean, local_module_package_index: str, modules_v2_dir: str
+) -> None:
+    """
+    Add a V2 module to an inmanta project using the `inmanta module add` command.
+    """
+    # Ensure that version 1.1.1, 1.2.0 and 1.2.3 exist in the pip index
```

Comment:
The comment versions do not match the actual versions.
```suggestion
    # Ensure that version 1.1.1 and 1.2.0 exist in the pip index
```